app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/inspections/AppEngineDeprecatedRuntimeInspection.java

```diff
@@ -104,16 +106,18 @@ public void applyFix(@NotNull Project project, @NotNull ProblemDescriptor descri
    * runtime tag, and the value is a deprecated Java runtime.
    */
   private boolean isAppEngineWebXmlDeprecatedRuntimeTag(XmlTag tag) {
-    XmlFile xmlFile = (XmlFile) tag.getContainingFile();
-    boolean isAppEngineWebXml =
-        xmlFile.getRootTag() != null
-            && AppEngineUtil.APP_ENGINE_WEB_XML_NAME.equals(xmlFile.getName())
-            && AppEngineUtil.APP_ENGINE_WEB_XML_ROOT_TAG_NAME.equals(
-                xmlFile.getRootTag().getName());
+    if (tag instanceof XmlFile) {
```

Comment:
Is this true of we want to check `getContainingFile`?

Reply:
sorry, I didn't understand?

Reply:
I mean the following snippet:
```
    if (tag instanceof XmlFile) {
      XmlFile xmlFile = (XmlFile) tag.getContainingFile();
```

we check `tag` for XmlFile, but then cast `tag.getContainingFile()`, I wonder if this is how it works, since it is not very obvious.

Reply:
oh you're right. thanks

Reply:
now that I think about it..if you look at the stacktrace you can see that its happening on a JS file. I wonder if we can reproduce with a JS file that contains XML snippet inside..

Reply:
fixed the cast check